feature/gnmi/ate_tests/telemetry_npu_asic_counters_test/README.md

```diff
@@ -0,0 +1,176 @@
+        * /packet/queueing-block/out-pkts,
+        * /packet/queueing-block/out-bytes
+    *   Define variance = 0.98.
+    *   Start traffic.
+    *   Wait for 30 secs.
+    *   Capture the out-pkts and out-bytes leaf values (2).
+    *   Calculation of step 2 - step 1.
+        *  30 * pps = out-pkts.
+        *  30 * pps * 300 = out-bytes.
+        *  Capture tgen rx packets, tgen rx packets/step(2) values >= variance.
+    *   Stop traffic.
+
+
+*   Drop Counters - Lookup-Block
+
+    *   Subscribe to the following, and capture the leaf values(1)
+        * /drop/lookup-block/no-route
+        * /drop/lookup-block/no-nexthop
+        * /drop/lookup-block/invalid-packet
+        * /drop/lookup-block/forwarding-policy (TBD)
+        * /drop/lookup-block/incorrect-software-state
+        * /drop/lookup-block/rate-limit (TBD)
+        * /drop/lookup-block/acl-drops
+    
+    *   Start traffic, verify traffic received on atePort2.
+    *   Shutdown ateport2. Wait for 30 secs.
+    *   Verify /drop/lookup-block/no-route counter populated (2).
+        * Step 2-Step 1(no-route), calculation 30 * pps.       
+    *   Unshut ateport2, verify traffic recovers on atePort2.
+
+    *   Configure static Null0 route for outerDst_IP1.
+    *   After 30 secs, Verify /drop/lookup-block/no-nexthop leaf populated(3).
+        * Step 3- Step 1(no-nexthop), calcuation 30 * pps.
+    *   Unconfigure static route, and verify traffic recovers on atePort2, stop traffic.
+
+    *   Send traffic with custom checksum in ipv4 header.
+    *   After 30 secs, Verify /drop/lookup-block/invalid-packet leaf populated(4).
+        * Step 4 - Step 1(invalid-packet), calculation 30*pps.
+    *   Stop traffic. 
+
+    *   /drop/lookup-block/forwarding-policy <TBD>
+
+    *   Send traffic to Outer_Dst_IP1, and verify traffic received on atePort2.
+    *   Remove ipv4 address on dutport1.
+    *   After 30 secs, Verify /drop/lookup-block/incorrect-software-state populated(5).
+        * Step 5 - Step 1(incorrect-software-state), calculation 30*pps.
+    *   Stop traffic.
+
+    *   /drop/lookup-block/rate-limit (TBD).
+
+    *   Configure ACL permit for source ip OuterSrcIP_2 and apply on egress DUT port-2.
+    *   Start traffic and wait for 30 secs.
+    *   Capture /drop/lookup-block/acl-drops(6).
+    *   Calculation - Tgen drop counter = Step 6 - Step 1.
+    *   Stop traffic.
+    *   Remove acl and verify traffic received on ATE port2.
+
+*   Drop Counters - Interface Block
+    *   drop/interface-block/state/oversubscription (TBD)
+
+*   Drop Counters - Queueing Block
+    *   Connect a gRIBI client to the DUT, make it become leader, flush and inject the following,
+
+        NHG#1 --> NH#1 {next-hop: ATEPort2IP}
+    *   Subscribe to /drop/queueing-block/oversubscription, and capture the leaf value(1).
+    *   Send traffic, 
+        * Flow 1 from ATE port 1 to ATE port 2 with line rate 75%.
+        * Flow 2 from ATE port 3 to ATE port 2 at 75% linerate.
```

Comment:
This should not work.
The OpenConfig states `Number of packets dropped due to oversubscription of the integrated-circuit subsystem block.` Not interface oversubscription.
Depending how we define "Queueng Block" but I would say that **block** capacity would be rather >> 100GE !
One possible scenario is oversubscription of HBM I/O (2Tbps in most ASICS). But it may be more complicated due to On-Chip/Off-Chip buffers and "slicing" of chip. 

I think right test would be to generate ingress traffic ~ 50% of chip capacity on single chip and direct it to single egress interfaces. Base on my understanding of HBM I/O this should overload it and cause drops.

Reply:
The oversubscription is per VOQ, which is per egress interface.
We can verify per VOQ traffic getting evicted to HBM and verify it.
That is my understanding,  I can check and get back.

Reply:
Even if a single interface is oversubscribed, we have to evict the packets to HBM and we will get VOQ drops on the ingress, since the oq scheduler on the egress will not provide grants.
The egress interface can be any where on the system on different NP, slice , different line card, but  the buffering will happen on ingress NP.